### DISCUSSION TITLE
Error when viewing source of previous page revision

### DISCUSSION THREAD

**Discussion by Chrisc-c-c** [requarks/wiki#3297] [Error / Bug Report] [answered]
**Describe the bug**
When trying to view the source of a previous version of a page, the source fails to load and I instead receive the following error page:

Oops, something went wrong...
/wiki/server/views/source.pug:7 5| block body 6| #root > 7| page-source( 8| :page-id=page.id 9| locale=page.localeCode 10| path=page.path The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined

**To Reproduce**
1. Open a wiki page.
2. Click on History.
3. Click the three dots next to a page revision.
4. Click "View Source".

**Expected behavior**
Revision source should be shown

**Host Info (please complete the following information):**
 - OS: Docker
 - Wiki.js version: 2.5.170
 - Database engine: Postgres 11.7

**Comment by mihailim** [comment]
Confirming, this happens with the non-Docker version as well.

* OS: Ubuntu 18.04
* Wiki.js version: 2.5.170
* Database engine: Postgres 13.1

**Comment by slysiou** [comment]
I get the same issue:
`/wiki/server/views/source.pug:7 5| block body 6| #root > 7| page-source( 8| :page-id=page.id 9| locale=page.localeCode 10| path=page.path The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined`

**Comment by mihailim** [comment]
This still happens and is still reproducible in the same way with 2.5.191.

**Reply by Chrisc-c-c** [reply]
Seems like the vast majority of bug reports have been closed without comment, but with a label of "migrate" attached. I wonder if this suggests the issue doesn't exist in version 3. 

It would be nice to have some clarification on this @NGPixel.

**Reply by NGPixel** [reply]
@Chrisc-c-c Issues were simply converted to Discussions for easier management. Nothing was closed...

**Reply by Chrisc-c-c** [reply]
Oh right, thanks for the clarification. The confusion arose from the fact that GitHub sends out emails letting everyone know that the issue "has been closed" at the point of the conversion.

Is the process going forward to raise bug reports within discussions, rather than the "issues" section then?

**Reply by NGPixel** [reply]
Correct. Issues were getting out of hand and it was impossible to manage. Discussions are now the way to go. Once validated as a reproducible bug, an issue will be created and linked to the discussion.

**Reply by Chrisc-c-c** [reply]
Great, thanks for clarifying.

**Comment by mihailim** [comment]
This still happens and is still reproducible in the same way with 2.5.194.

To clarify: In order to reproduce, you must attempt to view the page source of a _previous_ revision, not the current one (i.e. not the first one at the top of the list of revisions.) The current one works fine.

**Comment by PythonCoderAS** [comment]
This error still exists. It makes it really hard to view previous versions.

**Comment by mihailim** [comment]
This still happens exactly the same way with 2.5.214, regardless of node.js version, too.

**Comment by mihailim** [comment]
Same with 2.5.268, node 16.

**Comment by NGPixel** [comment, marked as answer]
Fixed by 92fe9d3e31b38c989a173e9afbbb5f6093095855

**Comment by mihailim** [comment]
Confirming, the patch solves the issue. Thank you so much!
